### PR TITLE
Feature/improve smoothline chart

### PR DESCRIPTION
**Pull request by AmauryLiet**
Thank you for contributing a pull request. 

Please ensure that you have signed the [CLA](https://docs.google.com/forms/d/19LpBBjykHPox18vrZvBbZUcK6gQTj7qv1O5hCduAZFU/viewform).

- [X] I have signed the CLA

Submitted on behalf of third-party: [kraynel](https://github.com/kraynel).

### CODE REVIEW

**Comment by marzolfb** [comment]
This looks like a great contribution and I'm reviewing it in more detail now. One thing that I think would be a fantastic addition to this pull request would be showing an example of these new options by either adding a new chart or modifying an existing one in the example app.

**Comment by marzolfb** [comment]
I think this is a wonderful addition and I'm happy to approve and merge it as-is but I'm going to give you a little time to see if you want to add any modifications to the example app to demonstrate these new option additions. I like that you allow for function inputs in the options you've added. Thanks!

**Comment by AmauryLiet** [comment]
Thanks for the comments!
Adding an example app in the repo seems like a great idea. I'd be happy to do, I may take a few days to add it since I'm lacking time right now, is it ok for you to wait before merging?

**Comment by marzolfb** [comment]
Yes, I can certainly wait. There's already an example app in the example directory - a new chart can be added easily.

File: src/Line.js [hunk 1]
```diff
@@ -82,93 +81,154 @@ export default class LineChart extends Component {
...
+      margin: options.margin,
+    };
+
+    let showAreas = typeof this.props.options.showAreas !== 'undefined'
+      ? this.props.options.showAreas
+      : true;
+    let strokeWidth = typeof this.props.options.strokeWidth !== 'undefined'
+      ? this.props.options.strokeWidth
+      : '1';
+    let strokeDasharray = typeof this.props.options.strokeDasharray !== 'undefined'
```
Comment (Minishlink) [inline src/Line.js]: strokeDashArray?

**Comment by kraynel** [comment]
Thank you @AmauryLiet for this PR, I did not take the time to do it myself at the time. I also signed the CLA.

**Comment by marzolfb** [comment]
@AmauryLiet Just checking back in - have you had a chance to add an example?

**Comment by AmauryLiet** [comment]
@marzolfb Thanks for the reminder, I'll be pushing the changes soon today

**Comment by AmauryLiet** [comment]
Hi @marzolfb, finally added an example to the PR where I expose some possible use of the modifications

**Comment by marzolfb** [comment]
My sincere apologies for the lengthy delay in reviewing this.

Fantastic work. The example is great. Thank you very much.

I'll merge this in for now but I may wait to release and update to npm so I can pull in the latest versions of dependencies to stay current.

**Pull request merged**